src/React InstantSearch/src/App.js

```diff
@@ -22,15 +47,25 @@ class App extends Component {
             <RefinementList attributeName="brand" />
             <Hits hitComponent={Hit} />
           </div>
-          <div style={{textAlign: 'center'}}><Pagination /></div>
+          <div style={{ textAlign: 'center' }}>
+            <Pagination />
+          </div>
         </InstantSearch>
       </div>
     );
   }
 }
 
 function Hit(props) {
-  return <div><Highlight attributeName="name" hit={props.hit} /></div>;
+  return (
+    <div>
+      <Highlight attributeName="name" hit={props.hit} />
+    </div>
+  );
 }
 
+Hit.propTypes = {
+  hit: PropTypes.optionalArrayOf(PropTypes.object),
```

Comment:
`optionalArrayOf` is not a valid type should be `arrayOf(xxx)` (see [docs](https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes)). You can also add `.isRequired` since the `hits` is always provided to the component.